### ISSUE TITLE
Deploy csi-driver

### ISSUE THREAD

**Issue by simonhammes** [hetznercloud/kubernetes-dev-env#50]
### TL;DR

It would save me a few keystrokes each time if there was an option to deploy https://github.com/hetznercloud/csi-driver :)

### Expected behavior

https://github.com/hetznercloud/csi-driver would be deployed if a variable (e.g. `deploy_csi_driver`) is set to `true`.

I can create a PR if you'd be willing to accept it.

**Comment by apricote** [comment]
Sounds good :) So far we only added what we needed in hcloud-cloud-controller-manager and csi-driver. Installing the csi-driver might be interesting for when we use this in cluster-autoscaler.

Out of curiosity, where are you using `kubernetes-dev-env` that you need csi-driver?

**Comment by simonhammes** [comment]
> Sounds good :) So far we only added what we needed in hcloud-cloud-controller-manager and csi-driver. Installing the csi-driver might be interesting for when we use this in cluster-autoscaler.

Okay, I'll send a PR :)

> Out of curiosity, where are you using `kubernetes-dev-env` that you need csi-driver?

I'm using it for developing my own stuff and want to have a development environment that's comparable to a managed K8S service at some large cloud provider (without paying a stupid amount of money). I could have used https://github.com/vitobotta/hetzner-k3s, but this project already has a registry configured (don't know about hetzner-k3s) and _it just works_!

Or would you recommend a different solution that might better fit my needs?

But maybe I did not fully understand the intent of your question :sweat_smile:

**Comment by apricote** [comment]
When we wrote this we had the very specific use-case of developing our integrations in mind. To keep it simple the project is really only intended for very short lived clusters used in development & testing. When we created this project there was some discussion around people using it to deploy their long-lived clusters, which is why we added the disclaimers and warning to the README.

I had not considered that it can be used for (regular application) development by other people, but nice to hear that its working for you :)
